langkit/expressions/collections.py

```diff
@@ -747,6 +747,25 @@ def collection_get(self, collection, index, or_null):
     return result
 
 
+@auto_attr
+def to_iterator(self, array):
+    array_expr = construct(array)
+    check_source_language(
+        array_expr.type.is_array_type,
+        '.at prefix must be an array: got {} instead'.format(
```

Comment:
Please update the error message, and add a testcase for it (`properties/invalid_iterator`, to be similar to what’s done for other expressions)